cpp/oneapi/dal/table/backend/homogen_table_builder_impl.hpp

```diff
@@ -81,12 +82,37 @@ class homogen_table_builder_impl
 
     void copy_data(const void* data, std::int64_t row_count, std::int64_t column_count) override {
         check_copy_data_preconditions(row_count, column_count);
+
+#ifdef ONEDAL_DATA_PARALLEL
+        auto this_q = data_.get_queue();
+        if (this_q) {
+            ONEDAL_ASSERT(is_known_usm(data_));
+            return detail::memcpy_host2usm(*this_q,
+                                           data_.get_mutable_data(),
+                                           data,
+                                           data_.get_size());
+        }
+#endif
         detail::memcpy(detail::default_host_policy{},
                        data_.get_mutable_data(),
                        data,
                        data_.get_size());
     }
 
+    void copy_data(const dal::array<byte_t>& data) override {
+        const std::int64_t dtype_size = detail::get_data_type_size(dtype_);
+        const std::int64_t input_element_count = data.get_size() / dtype_size;
+        ONEDAL_ASSERT(input_element_count * dtype_size == data.get_size());
+
+#ifdef ONEDAL_DATA_PARALLEL
+        auto input_q = data.get_queue();
+        if (input_q) {
+            return copy_data(*input_q, data.get_data(), input_element_count, 1);
+        }
+#endif
+        copy_data(data.get_data(), input_element_count, 1);
+    }
```

Comment:
Use `__ONEDAL_IF_QUEUE__` and `__ONEDAL_IF_NO_QUEUE__`, see `dal/detail/common.hpp`, usage example 
https://github.com/oneapi-src/oneDAL/blob/80e130dae144cdc378acef095839bd0f92c742a9/cpp/oneapi/dal/test/engine/linalg/matrix.hpp#L525